### ISSUE TITLE
Cannot destructure property 'statusCode' of 'errors[error.toLowerCase(...)]' as it is undefined

### ISSUE THREAD

**Issue by neophob** [nextauthjs/next-auth#1354]
**Describe the bug**
I let my laptop sleep over the weekend, once picket up the laptop and saw this error

```
(node:98205) UnhandledPromiseRejectionWarning: TypeError: Cannot destructure property 'statusCode' of 'errors[error.toLowerCase(...)]' as it is undefined.
    at error (/github/o7-heroku-pg-aws-s3/frontend/node_modules/next-auth/dist/server/pages/error.js:55:5)
    at Object.error (/github/o7-heroku-pg-aws-s3/frontend/node_modules/next-auth/dist/server/pages/index.js:65:31)
    at /github/o7-heroku-pg-aws-s3/frontend/node_modules/next-auth/dist/server/index.js:222:29
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/github/o7-heroku-pg-aws-s3/frontend/node_modules/next-auth/dist/server/index.js:52:103)
    at _next (/github/o7-heroku-pg-aws-s3/frontend/node_modules/next-auth/dist/server/index.js:54:194)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

**Expected behavior**
No unhandled exception ;)

**Feedback**

* [X] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by balazsorban44** [comment]
Which version of next-auth do you use?

**Comment by neophob** [comment]
Thanks, Indeed I used 3.4.2

**Comment by Cure4Life** [comment]
I'm getting this error as well using version 3.5.1. I was testing throwing an error in the credentials provider authorize method instead of just returning NULL.

> (node:24653) UnhandledPromiseRejectionWarning: TypeError: Cannot destructure property 'statusCode' of 'errors[error.toLowerCase(...)]' as it is undefined.
	at error (/home/bill/web/htdocs/NextJS/auth-test/node_modules/next-auth/dist/server/pages/error.js:55:5)
at Object.error (/home/bill/web/htdocs/NextJS/auth-test/node_modules/next-auth/dist/server/pages/index.js:65:31)
at /home/bill/web/htdocs/NextJS/auth-test/node_modules/next-auth/dist/server/index.js:222:29
at Generator.next (<anonymous>)
	at asyncGeneratorStep (/home/bill/web/htdocs/NextJS/auth-test/node_modules/next-auth/dist/server/index.js:52:103)
	at _next (/home/bill/web/htdocs/NextJS/auth-test/node_modules/next-auth/dist/server/index.js:54:194)
	at runMicrotasks (<anonymous>)
		at processTicksAndRejections (internal/process/task_queues.js:93:5)
		(Use `node --trace-warnings ...` to show where the warning was created)
		(node:24653) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 7)
		(node:24653) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by balazsorban44** [comment]
could you verify that your `node_modules/next-auth/package.json` version is also 3.5.1, and that your lock file (either yarn or package-lock) also states 3.5.1?

If that is the case, please provide a reproduction so I can look into it.

**Comment by Cure4Life** [comment]
Hello,
All versions state 3.5.1

I am using the sample credentials code found here: https://next-auth.js.org/providers/credentials
I changed the authorize handler to always evaluate to false to test invalid logins. I also specifically wanted to see how it would behave by throwing a custom error message especially since this is where I would be querying the database for the user and any errors occurring during that process would also likely be thrown, so if that occurred I wanted to see what it would look like. Here is my [...nextauth].js file:

> 
	
    import NextAuth from 'next-auth'
    import Providers from 'next-auth/providers'

    export default NextAuth({
        providers: [
            Providers.Credentials({
                // The name to display on the sign in form (e.g. 'Sign in with...')
                name: 'Credentials',
                // The credentials is used to generate a suitable form on the sign in page.
                // You can specify whatever fields you are expecting to be submitted.
                // e.g. domain, username, password, 2FA token, etc.
                credentials: {
                    username: {label: "Username", type: "text", placeholder: "jsmith"},
                    password: {label: "Password", type: "password"}
                },
                async authorize(credentials) {
                    // Add logic here to look up the user from the credentials supplied

                    // force a failed attempt
                    const user = false

                    if (user) {
                        // Any object returned will be saved in `user` property of the JWT
                        return user
                    }
                    else {
                        // If you return null or false then the credentials will be rejected
                        //return null
                        // You can also Reject this callback with an Error or with a URL:
                         throw new Error('error message') // Redirect to error page
                        // throw '/path/to/redirect'        // Redirect to a URL
                    }
                }
            })
        ]
    })

When attempting to log in, the screen does not change and the error trace is displayed in the console rather than redirecting to an error page.

Thank you

**Comment by balazsorban44** [comment]
I'll check this out, thanks for the additional info!

**Comment by alextaymx** [comment]
I'm also having this error

(node:5736) UnhandledPromiseRejectionWarning: TypeError: Cannot destructure property 'statusCode' of 'errors[error.toLowerCase(...)]' as it is undefined.

**Comment by balazsorban44** [comment]
if not on the latest version, could you please upgrade and see if the error goes away?

**Comment by guilhermedeandrade** [comment]
I'm on version `3.13.2` and this is also happening to me:

```js
TypeError: Cannot destructure property 'statusCode' of 'errors[error.toLowerCase(...)]' as it is undefined.
```

**Comment by alextaymx** [comment]
my version is "next-auth": "^3.13.2",
I notice the problem is that i can't return Error with custom string in the authorize function . 
I need to use the default value such as return new Error("default").

**Comment by balazsorban44** [comment]
at this point I ask you to open a new issue and provide a reproduction, thanks!

**Comment by balazsorban44** [comment]
I think I finally found the problem and pushed a fix for it in #1700! (`3.14.6` has this)

@alextaymx You are right, we only have a small number of error types accepted by the /error page. But we shouldn't have crashed just because you set something else. Here is an explanation: https://github.com/nextauthjs/next-auth/issues/1699#issuecomment-817427385
